tests/ui/delegation/macro-inside-glob.rs

```diff
@@ -0,0 +1,26 @@
+//@ check-pass
+
+#![feature(fn_delegation)]
+#![allow(incomplete_features)]
+
+trait Trait {
+    fn foo(&self) -> u8 { 0 }
+    fn bar(&self) -> u8 { 1 }
+}
+
+impl Trait for u8 {}
+
+struct S(u8);
+
+// Macro expansion works inside delegation items.
+macro_rules! u8 { () => { u8 } }
+macro_rules! self_0 { () => { &self.0 } }
+impl Trait for S {
+    reuse <u8!() as Trait>::* { self_0!() }
```

Comment:
I'm surprised to see `self_0` having access to the local binding `self`. Isn't this breaking hygiene rules? For comparison with normal functions, see [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=fa3411cc272fb4b2928a8d858241c889).

Reply:
Yeah, looks like a hygiene bug.
We'll need to decide what the "expected" context for `self` should be inherited from (since there's no explicit definition for it, like in regular methods).
Perhaps from `*`, but we don't keep the `*`'s precise span right now.

Reply:
This is a pre-existing issue with the way in which the `self` binding is created, not related to glob delegation, I'll fix this separately.

Reply:
PR - https://github.com/rust-lang/rust/pull/126497